addons/hr_holidays/models/hr_leave.py

```diff
@@ -614,7 +614,7 @@ def _compute_can_approve(self):
     def _compute_can_cancel(self):
         now = fields.Datetime.now()
         for leave in self:
-            leave.can_cancel = leave.id and leave.employee_id.user_id == self.env.user and leave.state == 'validate' and leave.date_from and leave.date_from > now
+            leave.can_cancel = leave.id and leave.employee_id.user_id == self.env.user and leave.state == 'validate' and leave.date_from and leave.date_from.date() >= now.date()
```

Comment:
instead, on line 615, we can have now = fields.Datetime.now().date() .. so that we don't have to convert datetime to date in a loop